### PR TITLE
DB Migration 2.5

### PR DESCRIPTION
**Pull request by matdenkas**
Steps to test: 
1. Delete your local DB volume.
2. Build and run `docker exec semaphore-core python3 tools/migrate_db.py` to test update to 2.5, validate with PGAdmin that rows are correct. (or don't im not your boss)
3. Edit target_version.json: 2.5->2.3 and rebuild.
4. Run `docker exec semaphore-core python3 tools/migrate_db.py` to test rollback, validate with PGAdmin that rows are correct. (or don't im not your boss)

### CODE REVIEW

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import csv
+
+#Constants
+CSV_FILE_PATHS = './tools/DatabaseMigration/2_5/init_data'
+
+
+class Migrator(IDatabaseMigration):
+
+    def update(self, databaseEngine: Engine) -> bool:
+        """This function updates the database to version 2.2 which adds rows to the locations reference
```
Comment (lovelysandlonelys) [inline tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py:28]: function description says we're updating to version 2.2 🤩

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                                    .where(table.c.notes == old['notes'])
+                                    .returning(table)
+                                    .values(new)
+                                    )
+                results.append(cursor.fetchall())
+            conn.commit() # Commit only after every update is successful
+        return results
+
+    def rollback(self, databaseEngine: Engine) -> bool:
+        """This function rolls the database back to version 2.1 which involves removing the changes
```
Comment (lovelysandlonelys) [inline tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py:91]: Rollback function description says we're rolling back to 2.1 from 2.2

**Review by lovelysandlonelys — CHANGES_REQUESTED** [review]
Just documentation changes, everything else looks great and tests run perfectly. 
image

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py (hunk 1 above)
Comment (matdenkas) [reply tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py:28]: I DIDNT STEAL YOUR CODE

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py (hunk 2 above)
Comment (matdenkas) [reply tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py:91]: Okay... maybe I stole your code

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py (hunk 1 above)
Comment (matdenkas) [reply tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py]: Fixed!

File: tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py (hunk 2 above)
Comment (matdenkas) [reply tools/DatabaseMigration/2_5/2_5_DatabaseMigration.py]: Fixed!

**Review by lovelysandlonelys — APPROVED** [review]

**Review by abeasley1722 — APPROVED** [review]
LGTM

File: tools/DatabaseMigration/2_4/2_4_DatabaseMigration.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import csv
+
+#Constants
+CSV_FILE_PATHS = './tools/DatabaseMigration/2_5/init_data'
+
+
+class Migrator(IDatabaseMigration):
+
+    def update(self, databaseEngine: Engine) -> bool:
+        """This function updates the database to version 2.5 which update Series descriptions
```
Comment (abeasley1722) [inline tools/DatabaseMigration/2_4/2_4_DatabaseMigration.py:28]: 2.4*

**Pull request merged**